observation_portal/common/rise_set_utils.py

```diff
@@ -80,10 +81,26 @@ def get_rise_set_intervals(request_dict, site=''):
     intervals_by_site = get_rise_set_intervals_by_site(request_dict)
     intervalsets_by_telescope = intervals_by_site_to_intervalsets_by_telescope(intervals_by_site, telescope_details.keys())
     filtered_intervalsets_by_telescope = filter_out_downtime_from_intervalsets(intervalsets_by_telescope)
-    filtered_intervalset = Intervals().union(filtered_intervalsets_by_telescope.values())
-    filtered_intervals = filtered_intervalset.toTupleList()
+    filtered_intervals_by_site = intervalsets_by_telescope_to_intervals_by_site(filtered_intervalsets_by_telescope)
 
-    return filtered_intervals
+    return filtered_intervals_by_site
+
+
+def intervalsets_by_telescope_to_intervals_by_site(intervalsets_by_telescope: dict) -> dict:
+    """Convert rise_sets ordered by telescope to rise_set ordered by site. Also convert from intervalset
+        to datetime tuple lists per site.
+    :param intervalsets_by_telescope:
+    :return: intervals by site
+    """
+    intervals_by_site = {}
```

Comment:
I don't know what the structure of an `intervalset` is, but you might be able to use a defaultdict here- something like:

```
from collections import defaultdict
...
def intervalsets_by_telescope_to_intervals_by_site(intervalsets_by_telescope: dict) -> dict:
    intervals_by_site = defaultdict(set)  # if an intervalset is of type set
    for telescope, intervalset in intervalsets_by_telescope.items():
        site = telescope.split('.')[2]
        intervals_by_site[site].update(intervalset)
    return {site: intervalset.toTupleList() for site, intervalset in intervals_by_site.items()}
```